sinks/kafka/driver.go

```diff
@@ -0,0 +1,166 @@
+type kafkaSink struct {
+	producer        kafka.Producer
+	timeSeriesTopic string
+	eventsTopic     string
+	sinkBrokerHosts []string
+}
+
+// START: ExternalSink interface implementations
+
+func (self *kafkaSink) Register(mds []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (self *kafkaSink) Unregister(mds []sink_api.MetricDescriptor) error {
+	return nil
+}
+
+func (self *kafkaSink) StoreTimeseries(timeseries []sink_api.Timeseries) error {
+	if timeseries == nil || len(timeseries) <= 0 {
+		return nil
+	}
+	for index := range timeseries {
+		err := self.produceKafkaMessage(timeseries[index], self.timeSeriesTopic)
+		if err != nil {
+			glog.Errorf("failed to produce Kafka messages : %s", err)
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
+		}
+	}
+	return nil
+}
+
+func (self *kafkaSink) StoreEvents(events []kube_api.Event) error {
+	if events == nil || len(events) <= 0 {
+		return nil
+	}
+	for index := range events {
+		err := self.produceKafkaMessage(events[index], self.eventsTopic)
+		if err != nil {
+			glog.Errorf("failed to produce Kafka messages: %s", err)
+			return fmt.Errorf("failed to produce Kafka messages: %s", err)
+		}
+	}
+	return nil
+}
+
+// produceKafkaMessage produces messages to kafka
+func (self *kafkaSink) produceKafkaMessage(v interface{}, topic string) error {
+	if v == nil {
+		return nil
+	}
+	jsonItems, err := json.Marshal(v)
+	if err != nil {
+		glog.Errorf("failed to transform the items to json : %s", err)
+		return fmt.Errorf("failed to transform the items to json : %s", err)
+	}
+	message := &proto.Message{Value: []byte(string(jsonItems))}
+
+	if err != nil {
+		glog.Errorf("failed to transform json to messages: %s", err)
+		return fmt.Errorf("failed to transform json to messages: %s", err)
+	}
+	_, err = self.producer.Produce(topic, partition, message)
+	if err != nil {
+		glog.Errorf("failed to produce message to %s:%d: %s", topic, partition, err)
+		return fmt.Errorf("failed to produce message to %s:%d: %s", topic, partition, err)
+	}
+	return nil
+}
+
+func (self *kafkaSink) DebugInfo() string {
+	info := fmt.Sprintf("%s\n", self.Name())
+	info += fmt.Sprintf("There are two kafka's topics: %s,%s:\n", self.eventsTopic, self.timeSeriesTopic)
+	info += fmt.Sprintf("The kafka's broker list is: %s", self.sinkBrokerHosts)
+	return info
+}
+
+func (self *kafkaSink) Name() string {
+	return "Apache-Kafka Sink"
+}
+
+func init() {
+	extpoints.SinkFactories.Register(NewKafkaSink, "kafka")
+}
+
+func NewKafkaSink(uri *url.URL, _ extpoints.HeapsterConf) ([]sink_api.ExternalSink, error) {
+	var kafkaSink kafkaSink
+	opts, _ := url.ParseQuery(uri.RawQuery)
+	if len(opts["timeseriestopic"]) >= 1 && len(opts["eventstopic"]) >= 1 {
+		kafkaSink.timeSeriesTopic = opts["timeseriestopic"][0]
+		kafkaSink.eventsTopic = opts["eventstopic"][0]
+	} else {
+		glog.Errorf("There is no topic assign for config kafka-sink")
+		return nil, fmt.Errorf("There is no topic assign for config kafka-sink")
+	}
```

Comment:
See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow - the else is unnecessary.